L1Trigger/Phase2L1ParticleFlow/src/BJetId.cc

```diff
@@ -0,0 +1,95 @@
+#include "DataFormats/Math/interface/deltaPhi.h"
+#include <cmath>
+
+static constexpr unsigned int n_particles_max = 10;
+
+BJetId::BJetId(const std::string &iInput, const BJetTFCache *cache, const std::string &iWeightFile, int iNParticles) {
+  NNvectorVar_.clear();
+  session_ = tensorflow::createSession(cache->graphDef.get());
+  fNParticles_ = iNParticles;
+
+  fPt_ = std::make_unique<float[]>(fNParticles_);
+  fEta_ = std::make_unique<float[]>(fNParticles_);
+  fPhi_ = std::make_unique<float[]>(fNParticles_);
+  fId_ = std::make_unique<float[]>(fNParticles_);
+  fCharge_ = std::make_unique<int[]>(fNParticles_);
+  fDZ_ = std::make_unique<float[]>(fNParticles_);
+  fDX_ = std::make_unique<float[]>(fNParticles_);
+  fDY_ = std::make_unique<float[]>(fNParticles_);
+  fInput_ = iInput;
+}
+
+BJetId::~BJetId() { tensorflow::closeSession(session_); }
+void BJetId::setNNVectorVar() {
+  NNvectorVar_.clear();
+  for (int i0 = 0; i0 < fNParticles_; i0++) {
+    if (fPt_.get()[i0] == 0) {
+      for (int i1 = 0; i1 < 13; i1++)
+        NNvectorVar_.push_back(0);
+      continue;
+    }
+    NNvectorVar_.push_back(fId_.get()[i0] == l1t::PFCandidate::Electron && fCharge_.get()[i0]<0);       // Electron
+    NNvectorVar_.push_back(fId_.get()[i0] == l1t::PFCandidate::Electron && fCharge_.get()[i0]>0);       // Positron
+    NNvectorVar_.push_back(fId_.get()[i0] == l1t::PFCandidate::Muon && fCharge_.get()[i0]<0);           // Muon
+    NNvectorVar_.push_back(fId_.get()[i0] == l1t::PFCandidate::Muon && fCharge_.get()[i0]>0);           // Anti-Muon
+    NNvectorVar_.push_back(fId_.get()[i0] == l1t::PFCandidate::Photon && fCharge_.get()[i0]==0);         // Photon
+    NNvectorVar_.push_back(fId_.get()[i0] == l1t::PFCandidate::NeutralHadron && fCharge_.get()[i0]==0);  // Neutral Had
+    NNvectorVar_.push_back(fId_.get()[i0] == l1t::PFCandidate::ChargedHadron && fCharge_.get()[i0]<0);  // Pion
+    NNvectorVar_.push_back(fId_.get()[i0] == l1t::PFCandidate::ChargedHadron && fCharge_.get()[i0]>0);  // Anti-Pion
+    NNvectorVar_.push_back(fDZ_.get()[i0]);   //dZ
+    NNvectorVar_.push_back(std::sqrt(fDX_.get()[i0]*fDX_.get()[i0] + fDY_.get()[i0]*fDY_.get()[i0]));  //d0
+    NNvectorVar_.push_back(fPt_.get()[i0]);   //pT as a fraction of jet pT
+    NNvectorVar_.push_back(fEta_.get()[i0]);  //dEta from jet axis
+    NNvectorVar_.push_back(fPhi_.get()[i0]);  //dPhi from jet axis
+  }
+}
+float BJetId::EvaluateNN() {
+  tensorflow::Tensor input(tensorflow::DT_FLOAT,
+                           {1, (unsigned int)NNvectorVar_.size(), 1});
+  for (unsigned int i = 0; i < NNvectorVar_.size(); i++) {
+    input.tensor<float,3>()(0, i, 0) = float(NNvectorVar_[i]);
+  }
+  std::vector<tensorflow::Tensor> outputs;
+  tensorflow::run(session_, {{fInput_, input}}, {"sequential/dense_2/Sigmoid"}, &outputs);
+  //tensorflow::run(session_, {{fInput_, input}}, {"sequential/sigmoid/Sigmoid"}, &outputs);//TODO: for QKeras models, add auto switch
```

Comment:
Shouldn't the `"sequential/dense_2/Sigmoid"` string be a configurable parameter, set together with the training file? (unless it can be auto-derived from the graph object itself)